BarcodeParserBuilder/Barcodes/CODE128/Code128Barcode.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using BarcodeParserBuilder.Infrastructure;
+using BarcodeParserBuilder.Exceptions;
+
+namespace BarcodeParserBuilder.Barcodes.CODE128
+{
+    public class Code128Barcode : Barcode
+    {
+
+        public Code128Barcode() : base() { }
+
+        public Code128Barcode(Code128SymbologyIdentifier symbologyIdentifier) : base(symbologyIdentifier) { }
+
+        public override ProductCode? ProductCode
+        {
+            get => (ProductCode?)BarcodeFields[nameof(ProductCode)].Value;
+            set => BarcodeFields[nameof(ProductCode)].SetValue(value);
+        }
+
+        protected override FieldCollection BarcodeFields { get; } = new()
+        {
+            new BarcodeField<ProductCode>(BarcodeType.CODE128, nameof(ProductCode), 2, 48)
```

Comment:
does the Code128 have a maximum length? is it 48 characters?

Reply:
Not strictly, it is limited by the ability of the scanner. But as far as I read, all major vendors say that the maximum which should be supported safely is 44, some support 48. So the Code128 barcode should not exceed 48, otherwise things may go wrong.

Reply:
then this is a good limitation i think. if a barcode is broken, it should be put in question why and stuff if there are scanner limitations etc

Reply:
Yes, I set the limitation also on reason that the BarcodeParserBuilder actually does not know whether the reading came from real scanner (which has the hardware limitation) or from some other (possibly faulty/malicious feed). So I thought it should behave according to realistic situation.